code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1717,6 +1717,12 @@ func (sc *StorageSmartContract) finishAllocation(
 
 	alloc.Finalized = true
 
+	challenges, err := sc.getAllocationChallenges(alloc.ID, balances)
+	if err != nil {
+		return err
+	}
+
+	emitUpdateChallengesForExpiredAllocations(alloc.ID, balances, challenges)
```

Comment:
We need to update the open challenge stats in MPT too, such as the [Stats](https://github.com/0chain/0chain/blob/6c19a04429c3cdeb9dfb08a7c150df4fc06ef6c7/code/go/0chain.net/smartcontract/storagesc/models.go#L630) in Allocation, and the [BlobberRewardStat](https://github.com/0chain/0chain/blob/0b6013e510c26e04f39b80ce61cdaf0a60b4b6a1/code/go/0chain.net/smartcontract/storagesc/challenge.go#L664) in ongoing partitions if any, etc.

Reply:
In the StorageAllocationStats in MPT, I can't think of a problem but will it really make some useful difference?

Reply:
The data in event db and MPT should not be different, event db will provide richer info but should be aligned with the MPT data. And it should be applied on all cases in processing event db.

Reply:
@dabasov Should I update it according to MPT? 

it should be applied on all cases in processing event db. But why is it mandatory for expired allocations?

Reply:
Updated. Please review again. Thanks!!